Bug: Annotated optional query parameter inconsistency

### Description

Annotated optional query parameters are inconsistent compared to non annotated and required query parameters.

### URL to code causing the issue

_No response_

### MCVE

```python
from typing import Annotated

from litestar import Litestar, get
from litestar.params import Parameter


@get("/optional", sync_to_thread=False)
def optional(key: str | None = None) -> dict[str, str | None]:
    return {"key": key}


@get("/annotated_required", sync_to_thread=False)
def annotated_required(param: Annotated[str, Parameter(query="key")]) -> dict[str, str]:
    return {"key": param}


@get("/annotated_optional", sync_to_thread=False)
def annotated_optional(
    param: Annotated[str | None, Parameter(query="key")] = None
) -> dict[str, str | None]:
    return {"key": param}


app = Litestar([optional, annotated_required, annotated_optional])
```


### Steps to reproduce

```bash
# optional - non annotated
$ curl -w '\n' -s 'localhost:8000/optional?param=a'
{"key":null}
$ curl -w '\n' -s 'localhost:8000/optional?key=a'
{"key":"a"}

# annotated - non optional
$ curl -w '\n' -s 'localhost:8000/annotated_required?param=a'
{"status_code":400,"detail":"Missing required query parameter 'key' for url http://localhost:8000/annotated_required?param=a"}
$ curl -w '\n' -s 'localhost:8000/annotated_required?key=a'
{"key":"a"}

# annotated - optional
$ curl -w '\n' -s 'localhost:8000/annotated_optional?param=a'
{"key":"a"}
$ curl -w '\n' -s 'localhost:8000/annotated_optional?key=a'
{"key":null}
```
```


### Screenshots

```bash
"![SCREENSHOT_DESCRIPTION](SCREENSHOT_LINK.png)"
```


### Logs

_No response_

### Litestar Version

2.2.1

### Platform

- [ ] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

<!-- POLAR PLEDGE BADGE START -->
---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh Litestar dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/2514">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2514/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/2514/pledge.svg">
</picture>
</a>
<!-- POLAR PLEDGE BADGE END -->

Hmm

given:

```py
    @get("/optional-annotated-default")
    def handle_default_annotated(
        param: Annotated[Optional[str], Parameter(query="key")] = None
    ) -> Dict[str, Optional[str]]:
        return {"key": param}
```
then:

```py
>       assert optional_default_client.get("/optional-annotated-default", params={"key": "a"}).json() == {"key": "a"}
E       AssertionError: assert {'key': None} == {'key': 'a'}
E         Differing items:
E         {'key': None} != {'key': 'a'}
E         Use -v to get more diff
```
Seems like a value can't be provided at all for the optional/annotated/with default case..

--- a/tests/unit/test_params.py
+++ b/tests/unit/test_params.py
@@ -1,4 +1,4 @@
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, Generator, List, Optional
 
 import pytest
 from typing_extensions import Annotated
@@ -8,7 +8,7 @@ from litestar.di import Provide
 from litestar.exceptions import ImproperlyConfiguredException
 from litestar.params import Body, Dependency, Parameter
 from litestar.status_codes import HTTP_200_OK, HTTP_201_CREATED, HTTP_400_BAD_REQUEST, HTTP_500_INTERNAL_SERVER_ERROR
-from litestar.testing import create_test_client
+from litestar.testing import TestClient, create_test_client
 
 
 def test_parsing_of_parameter_as_annotated() -> None:
@@ -219,3 +219,78 @@ def test_regex_validation() -> None:
 
         response = client.get("/val_regex?text=c")
         assert response.status_code == HTTP_400_BAD_REQUEST
+
+
+@pytest.fixture(name="optional_no_default_client")
+def optional_no_default_client_fixture() -> Generator[TestClient, None, None]:
+    @get("/optional-no-default")
+    def handle_optional(key: Optional[str]) -> Dict[str, Optional[str]]:
+        return {"key": key}
+
+    @get("/optional-annotated-no-default")
+    def handle_optional_annotated(param: Annotated[Optional[str], Parameter(query="key")]) -> Dict[str, Optional[str]]:
+        return {"key": param}
+
+    with create_test_client(route_handlers=[handle_optional, handle_optional_annotated]) as client:
+        yield client
+
+
+def test_optional_query_parameter_consistency_no_default_queried_without_param(
+    optional_no_default_client: TestClient,
+) -> None:
+    assert optional_no_default_client.get("/optional-no-default", params={}).json() == {"key": None}
+    assert optional_no_default_client.get("/optional-annotated-no-default", params={}).json() == {"key": None}
+
+
+def test_optional_query_parameter_consistency_no_default_queried_with_expected_param(
+    optional_no_default_client: TestClient,
+) -> None:
+    assert optional_no_default_client.get("/optional-no-default", params={"key": "a"}).json() == {"key": "a"}
+    assert optional_no_default_client.get("/optional-annotated-no-default", params={"key": "a"}).json() == {"key": "a"}
+
+
+def test_optional_query_parameter_consistency_no_default_queried_with_other_param(
+    optional_no_default_client: TestClient,
+) -> None:
+    assert optional_no_default_client.get("/optional-no-default", params={"param": "a"}).json() == {"key": None}
+    assert optional_no_default_client.get("/optional-annotated-no-default", params={"param": "a"}).json() == {
+        "key": None
+    }
+
+
+@pytest.fixture(name="optional_default_client")
+def optional_default_client_fixture() -> Generator[TestClient, None, None]:
+    @get("/optional-default")
+    def handle_default(key: Optional[str] = None) -> Dict[str, Optional[str]]:
+        return {"key": key}
+
+    @get("/optional-annotated-default")
+    def handle_default_annotated(
+        param: Annotated[Optional[str], Parameter(query="key")] = None
+    ) -> Dict[str, Optional[str]]:
+        return {"key": param}
+
+    with create_test_client(route_handlers=[handle_default, handle_default_annotated]) as client:
+        yield client
+
+
+def test_optional_query_parameter_consistency_wiht_default_queried_without_param(
+    optional_default_client: TestClient,
+) -> None:
+    assert optional_default_client.get("/optional-default", params={}).json() == {"key": None}
+    assert optional_default_client.get("/optional-annotated-default", params={}).json() == {"key": None}
+
+
+def test_optional_query_parameter_consistency_with_default_queried_with_expected_param(
+    optional_default_client: TestClient,
+) -> None:
+    assert optional_default_client.get("/optional-default", params={"key": "a"}).json() == {"key": "a"}
+    assert optional_default_client.get("/optional-annotated-default", params={"key": "a"}).json() == {"key": "a"}
+
+
+def test_optional_query_parameter_consistency_with_default_queried_with_other_param(
+    optional_default_client: TestClient,
+) -> None:
+    assert optional_default_client.get("/optional-default", params={"param": "a"}).json() == {"key": None}
+    assert optional_default_client.get("/optional-annotated-default", params={"abc": "xyz"}).json() == {"key": None}
+    assert optional_default_client.get("/optional-annotated-default", params={"param": "a"}).json() == {"key": None}

--- a/tests/unit/test_params.py
+++ b/tests/unit/test_params.py
@@ -231,7 +231,7 @@ def optional_no_default_client_fixture() -> Generator[TestClient, None, None]:
     def handle_optional_annotated(param: Annotated[Optional[str], Parameter(query="key")]) -> Dict[str, Optional[str]]:
         return {"key": param}
 
-    with create_test_client(route_handlers=[handle_optional, handle_optional_annotated]) as client:
+    with create_test_client(route_handlers=[handle_optional, handle_optional_annotated], openapi_config=None) as client:
         yield client
 
 
@@ -270,7 +270,7 @@ def optional_default_client_fixture() -> Generator[TestClient, None, None]:
     ) -> Dict[str, Optional[str]]:
         return {"key": param}
 
-    with create_test_client(route_handlers=[handle_default, handle_default_annotated]) as client:
+    with create_test_client(route_handlers=[handle_default, handle_default_annotated], openapi_config=None) as client:
         yield client
 
 

